src/SFA.DAS.AANHub.Application/Apprentices/Commands/PatchApprenticeMember/PatchApprenticeMemberCommandHandler.cs

```diff
@@ -0,0 +1,57 @@
+﻿using MediatR;
+using System.Text.Json;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Domain.Entities;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using SFA.DAS.AANHub.Domain.Interfaces;
+using static SFA.DAS.AANHub.Domain.Common.Constants;
+
+namespace SFA.DAS.AANHub.Application.Apprentices.Commands.PatchApprenticeMember
+{
+    public class PatchApprenticeMemberCommandHandler : IRequestHandler<PatchApprenticeMemberCommand, ValidatedResponse<PatchApprenticeMemberCommandResponse>>
+    {
+        private readonly IApprenticesWriteRepository _apprenticesWriteRepository;
+        private readonly IAuditWriteRepository _auditWriteRepository;
+        private readonly IAanDataContext _aanDataContext;
+
+        public PatchApprenticeMemberCommandHandler(IApprenticesWriteRepository apprenticesWriteRepository,
+            IAuditWriteRepository auditWriteRepository,
+            IAanDataContext aanDataContext)
+        {
+            _apprenticesWriteRepository = apprenticesWriteRepository;
+            _auditWriteRepository = auditWriteRepository;
+            _aanDataContext = aanDataContext;
+        }
+
+        public async Task<ValidatedResponse<PatchApprenticeMemberCommandResponse>> Handle(PatchApprenticeMemberCommand command, CancellationToken cancellationToken)
+        {
+            var success = true;
+
+            var apprentice = await
+                _apprenticesWriteRepository.GetApprentice(command.ApprenticeId);
+
+            if (apprentice == null)
+                return new ValidatedResponse<PatchApprenticeMemberCommandResponse>((PatchApprenticeMemberCommandResponse)null!);
```

Comment:
I think here you need to return Success=false?

Reply:
I tried this earlier but `ValidatedResponse` did not like it. We can try and find a solution sometime in between in my personal view.

Reply:
That does not make sense, need to see what you mean.